### PR TITLE
feat: export all schemas and data at once in export tool

### PR DESCRIPTION
**Pull request by fengjiachun**
I hereby agree to the terms of the [GreptimeDB CLA](https://github.com/GreptimeTeam/.github/blob/main/CLA.md).

## What's changed and what's your intention?

1. export all schemas and data at once in export tool
2. remove `export table data` cmd, use `database data` instead.

## Checklist

- [ ] I have written the necessary rustdoc comments.
- [ ] I have added the necessary unit tests and integration tests.
- [x] This PR requires documentation updates.

### CODE REVIEW

File: src/cmd/src/cli/export.rs [hunk 1]
```diff
@@ -42,14 +42,13 @@ type TableReference = (String, String, String);
...
 enum ExportTarget {
-    /// Corresponding to `SHOW CREATE TABLE`
-    #[default]
+    /// Export all table schemas, corresponding to `SHOW CREATE TABLE`.
     CreateTable,
-    /// Corresponding to `EXPORT TABLE`
-    #[deprecated(note = "Please use `DatabaseData` instead.")]
-    TableData,
-    /// Corresponding to `EXPORT DATABASE`
+    /// Export all table data, corresponding to `COPY DATABASE TO`.
```
Comment (killme2008) [inline src/cmd/src/cli/export.rs:47]: It makes me more confused. Is it to export all the schemas and data in a database?

File: src/cmd/src/cli/export.rs [hunk 2]
```diff
@@ -42,14 +42,13 @@ type TableReference = (String, String, String);
...
     DatabaseData,
+    /// Export all table schemas and data at once.
```
Comment (killme2008) [inline src/cmd/src/cli/export.rs:49]: What's the difference with `database`?

**Review by killme2008 — COMMENTED** [review]
What's purpose of this tool? To export a database?

File: src/cmd/src/cli/export.rs [hunk 3]
```diff
@@ -42,14 +42,13 @@ type TableReference = (String, String, String);
 
 #[derive(Debug, Default, Clone, ValueEnum)]
 enum ExportTarget {
-    /// Corresponding to `SHOW CREATE TABLE`
-    #[default]
+    /// Export all table schemas, corresponding to `SHOW CREATE TABLE`.
```
Comment (killme2008) [inline src/cmd/src/cli/export.rs:45]: If I understand correctly, this tool is used to export data for a database. I think we can change all these three options into:
1. `schema`.
2. `data`
3. and `all`.

File: src/cmd/src/cli/export.rs (hunk 2 above)
Comment (fengjiachun) [reply src/cmd/src/cli/export.rs:49]: Schema:schemas, 
Data: data,
All: schemas&data

**Comment by fengjiachun** [comment]
> What's purpose of this tool? To export a database?

It can export a database (including schema and data), or it can export all databases under a catalog.

**Comment by fengjiachun** [comment]
@killme2008 PTAL

**Review by WenyXu — APPROVED** [review]
LGTM

**Review by killme2008 — COMMENTED** [review]
Do we have any tests for this tool?

**Comment by fengjiachun** [comment]
> Do we have any tests for this tool?

There is already a unit test for `export schema` 

https://github.com/GreptimeTeam/greptimedb/blob/1746b58cdf121c75a55d23da68f70709175716b7/src/cmd/src/cli/export.rs#L495

However, testing for `export data` might be challenging. Once we provide the import tool, this test should become easier.

**Comment by killme2008** [comment]
> > Do we have any tests for this tool?
> 
> There is already a unit test for `export schema`
> 
> https://github.com/GreptimeTeam/greptimedb/blob/1746b58cdf121c75a55d23da68f70709175716b7/src/cmd/src/cli/export.rs#L495
> 
> However, testing for `export data` might be challenging. Once we provide the import tool, this test should become easier.

Got，we need a `greptime cli import` command.

**Review by killme2008 — APPROVED** [review]
LGTM

**Pull request merged**